rewriter/TEnum.cc

```diff
@@ -245,6 +245,10 @@ void TEnum::run(core::MutableContext ctx, ast::ClassDef *klass) {
         auto sig = ast::MK::Sig0(klass->declLoc, std::move(return_type_ast));
         auto method = ast::MK::SyntheticMethod0(klass->loc, klass->declLoc, core::Names::serialize(),
                                                 ast::MK::RaiseTypedUnimplemented(klass->declLoc));
+        auto visibility =
+            ast::MK::Send0(klass->declLoc, ast::MK::Self(klass->declLoc), core::Names::public_(), klass->declLoc);
```

Comment:
Ah, never mind about the compiler.  You'll need to use `ast::MK::Send` here so that you can pass in flags that indicate that private sends are OK, like:

https://github.com/sorbet/sorbet/blob/6a1084aa558fbf334ccb605e40f347c6fa7f5069/ast/desugar/Desugar.cc#L673-L683

Sorry about the runaround!

Reply:
No worries at all! That's really great to know, thank you :) I'll fix it up first thing next week.